recipes/log4cxx/all/conanfile.py

```diff
@@ -0,0 +1,146 @@
+        elif tools.Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration("log4cxx requires a compiler that supports at least C++17")
+
+    def validate(self):
+        #todo: if compiler doesn't support C++17, boost can be used instead
+        self._validate_compiler_settings()
+
+    def source(self):
+        #OSError: [WinError 123] The filename, directory name, or volume label syntax is incorrect:
+        #'source_subfolder\\src\\test\\resources\\output\\xyz\\:'
+        pattern = "*[!:]"
+        tools.get(**self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True,
+                  pattern=pattern)
+
+    def _patch_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["BUILD_TESTING"] = False
+            if self.settings.os == "Windows":
+                self._cmake.definitions["LOG4CXX_INSTALL_PDB"] = False
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        self.copy("NOTICE", src=self._source_subfolder, dst="licenses")
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {"log4cxx": "log4cxx::log4cxx"}
+        )
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(module_file, targets):
+        content = ""
+        for alias, aliased in targets.items():
+            content += textwrap.dedent("""\
+                if(TARGET {aliased} AND NOT TARGET {alias})
+                    add_library({alias} INTERFACE IMPORTED)
+                    set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+                endif()
+            """.format(alias=alias, aliased=aliased))
+        tools.save(module_file, content)
+
+    @property
+    def _module_subfolder(self):
+        return os.path.join("lib", "cmake")
+
+    @property
+    def _module_file_rel_path(self):
+        return os.path.join(self._module_subfolder,
+                            "conan-official-{}-targets.cmake".format(self.name))
+
+    def package_info(self):
+        self.cpp_info.builddirs.append(self._module_subfolder)
+        self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
+        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
```

Comment:
```suggestion
        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
        self.cpp_info.build_modules["names"] = "liblog4cxx"
```

https://github.com/apache/logging-log4cxx/blob/rel/v0.12.0/liblog4cxx.pc.in

Reply:
You mean `self.cpp_info.names["pkg_config"] = "liblog4cxx"`

Reply:
Yes, thank you!